tables/__init__.py

```diff
@@ -8,25 +8,35 @@
 """
 import os
 from ctypes import cdll
+from ctypes.util import find_library
 import platform
 
 
 # Load the blosc2 library, and if not found in standard locations,
 # try this directory (it should be automatically copied in setup.py).
 current_dir = os.path.dirname(__file__)
 platform_system = platform.system()
-blosc2_lib = "libblosc2"
+blosc2_lib_hardcoded = "libblosc2"
 if platform_system == "Linux":
-    blosc2_lib += ".so"
+    blosc2_lib_hardcoded += ".so"
 elif platform_system == "Darwin":
-    blosc2_lib += ".dylib"
+    blosc2_lib_hardcoded += ".dylib"
 else:
-    blosc2_lib += ".dll"
-try:
-    cdll.LoadLibrary(blosc2_lib)
-except OSError:
-    cdll.LoadLibrary(os.path.join(current_dir, blosc2_lib))
-
+    blosc2_lib_hardcoded += ".dll"
+blosc2_found = False
+blosc2_search_paths = [blosc2_lib_hardcoded,
+                       os.path.join(current_dir, blosc2_lib_hardcoded),
+                       find_library("blosc2")]
```

Comment:
I see a few problems:
1. `find_library()` returns None if it doesn't find the library, and `cdll.LoadLibrary(None)` doesn't DTRT:
```pycon
>>> ctypes.util.find_library('missing.so') is None
True
>>> ctypes.cdll.LoadLibrary(None)
<CDLL 'None', handle 7f317dd372c0 at 0x7f316f9e7f10>
```

That loop needs an additional `if blosc2_lib` check.

2. Loading of library from $PWD creates the obvious security issue where an attacker can place a file in $PWD and cause the user to load it. Loading from $PWD can be done in a test environment, but MUST NOT be done in production code.

3. `find_library()` doesn't seem to fit the bill. Quoting the docs: "The purpose of the find_library() function is to locate a library in a way similar to what the compiler or runtime loader does (on platforms with several versions of a shared library the most recent should be loaded)". It would find the newest version of the library, which might or might not be what is needed. The whole point of the SONAME field (i.e. the thing that gives the .2 suffix) is to refer to a specific version from code using the library, so that when a future incompatible version appears, existing code is not broken.

Reply:
https://src.fedoraproject.org/rpms/python-tables/blob/rawhide/f/0007-Drop-misguided-check.patch
This is the patch I pushed in Fedora. It ignores non-Linux systems, so it's not suitable for inclusion upstream.